techTree.md

```diff
@@ -0,0 +1,97 @@
+> * **Yellow**: Support work needed for a major milestone, such as software tools or data analysis.
+> * **Green**: Nice-to-haves that are not critical but still valuable to Stateless Ethereum and/or Eth2. 
+
+
+
+## Witness Prototype
+
+It's possible to build a stateless client today with the current network primitives; beam sync is an example of a semi-stateless client implementation. The objective of this milestone is to have implemented clients on mainnet passing around prototype witnesses. It's likely that the first witnesses will be too large to propagate efficiently, but a first implementation will be a good starting point to improve upon through other objectives such as code merkleization and a binary state trie. 
+
+The starting point for a witness format prototype is the [Witness Specification](./witness.md), which aims to define witnesses without ambiguity for the sake of soundness in implementation. 
+
+The specification is still in draft form and contributions, especially in writing tests, are welcome. 
+
+In addition to the specification, there are two noteworthy issues that a witness implementation must 'solve' in some way.
+
+#### Witness Indexing
+
+We need a reliable way of determining which witness corresponds to which block and associated state. This could be as simple as putting a `witnessHash` field into the block header, or something else that serves the same purpose but in a different way.
+
+#### Stateless tx Validation
+
+This is an interesting early problem [thoroughly summarized on the ethresearch forums](https://ethresear.ch/t/the-current-transaction-validation-rules-require-access-to-the-account-trie/7046). In summary, clients need to quickly check if incoming transactions (waiting to be mined into a future block) are at least *eligible* to be included in a future block. 
+
+
+
+## EVM
+
+Stateless Ethereum will require a change to EVM semantics, particularly in gas execution and opcode repricing.  Because the EVM is currently an implicit (and not explicit) Ethereum State Transition Function, changes to semantics can be quite complex, and will likely affect future work on Eth1x as well as Eth2.
+
+#### Witness Gas Accounting
+
+Generating a block’s witness involves some computation that will be performed by the block’s miner, and therefore will need to have an associated gas cost, paid for by the transaction’s sender. Currently there is little more than a "finger in the air" estimate of what the gas costs for witness production should be. The problem requires substantive analysis of EVM execution and historical blocks. 
+
+An unfortunate side-effect of opcode repricing is potential breaking changes in smart contracts that have built-in assumptions about gas costs.  Various strategies have been proposed to mitigate this negative side effect, with discussion centered around making gas used *unobservable* in whole or in part. 
+
+>  **UNGAS vs Oil / Karma** 
```

Comment:
Why are UNGAS vs Oil/Karma in a indented gray block?

Reply:
Because for the moment I feel like Oil / Karma / UNGAS are all in a category of proposed solutions to the problem of gas repricing in the interest of witnesses. In this case, it's helpful to understand the problem in terms of solutions (because it's so abstract), but I wanted to distinguish them visually because they're all actively being discussed

Reply:
When reading, I thought that the indented gray box was strange and didn't understand its purpose or how it relates to the rest of the doc. Perhaps it could use a label and transition, like the following.

> **Aside.** With disagreement over the way forward at the moment, one proposed strategy worth mentioning is Oil / Karma / UNGAS....

Or maybe it is OK as it.